src/main/java/catalin/Reader.java

```diff
@@ -1,27 +1,23 @@
 package catalin;
 
 import java.io.*;
+import java.util.regex.MatchResult;
 
 public class Reader {
-    File file;
-    BufferedReader bufferedReader ;
+    File filename;
```

Comment:
Please remove all the fields from Reader.java and use local variables for each function.